### PR TITLE
[MM-15424] Update dependencies

### PR DESCRIPTION
**Pull request by migbot**
#### Summary
Updated babel dependencies to 7.5.5 and added core-js 3 so that the app builds with the core-js 3 update to mattermost-redux. I regenerated modulePaths as well given the changes to core-js directory structure.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15424

#### Device Information
This PR was tested on:
* Galaxy S7, Android 8.1
* iPhone 8, iOS 12.3

### CODE REVIEW

File: packager/moduleNames.js [hunk 1]
```diff
@@ -604,16 +1470,93 @@ module.exports = [
...
-    'node_modules/scheduler/cjs/scheduler-tracing.production.min.js',
-    'node_modules/scheduler/cjs/scheduler.production.min.js',
+    'node_modules/scheduler/cjs/scheduler-tracing.development.js',
+    'node_modules/scheduler/cjs/scheduler.development.js',
     'node_modules/scheduler/index.js',
     'node_modules/scheduler/tracing.js',
     'node_modules/semver/semver.js',
+    'node_modules/serialize-error/index.js',
+    'node_modules/shallow-equals/index.js',
+    'node_modules/socketcluster-client/index.js',
```
Comment (enahum) [inline packager/moduleNames.js]: remove socketcluster refs

File: packager/modulePaths.js [hunk 2]
```diff
@@ -610,7 +914,42 @@ module.exports = ['./node_modules/app/components/loading.js',
     './node_modules/node_modules/scheduler/index.js',
     './node_modules/node_modules/scheduler/tracing.js',
     './node_modules/node_modules/semver/semver.js',
+    './node_modules/node_modules/serialize-error/index.js',
+    './node_modules/node_modules/shallow-equals/index.js',
+    './node_modules/node_modules/socketcluster-client/index.js',
```
Comment (enahum) [inline packager/modulePaths.js]: remove from here too

**Comment by crspeller** [comment]
@migbot Is the cherry pick correct? Generally dep updates don't get backported.

**Comment by enahum** [comment]
Wait @migbot didn't you mentioned that you needed this to fix an issue on 1.22?

**Pull request merged**